pyaedt/edb_grpc/core/edb_data/hfss_simulation_setup_data.py

```diff
@@ -0,0 +1,2146 @@
+
+    @property
+    def via_density(self):
+        """Via density.
+
+        Returns
+        -------
+        float
+        """
+        return self._via_settings.ViaDensity
+
+    @via_density.setter
+    def via_density(self, value):
+        self._via_settings.ViaDensity = value
+        self._parent._update_setup()
+
+    @property
+    def via_material(self):
+        """Via material.
+
+        Returns
+        -------
+        str
+        """
+        return self._via_settings.ViaMaterial
+
+    @via_material.setter
+    def via_material(self, value):
+        self._via_settings.ViaMaterial = value
+        self._parent._update_setup()
+
+    @property
+    def via_num_sides(self):
+        """Via number of sides.
+
+        Returns
+        -------
+        int
+        """
+        return self._via_settings.ViaNumSides
+
+    @via_num_sides.setter
+    def via_num_sides(self, value):
+        self._via_settings.ViaNumSides = value
+        self._parent._update_setup()
+
+    @property
+    def via_style(self):
+        """Via style.
+        Options:
+        1- ``k25DViaWirebond``.
+        2- ``k25DViaRibbon``.
+        3- ``k25DViaMesh``.
+        4- ``k25DViaField``.
+        5- ``kNum25DViaStyle``.
+
+        Returns
+        -------
+        str
+        """
+        return self._via_settings.ViaStyle.ToString()
+
+    @via_style.setter
+    def via_style(self, value):
+        self._via_settings.ViaStyle = self._via_style_mapping[value]
+        self._parent._update_setup()
+
+
+class AdvancedMeshSettings(object):
+    """Manages EDB methods for advanced mesh settings."""
+
+    def __init__(self, parent):
+        self._parent = parent
+
+    @property
+    def _advanced_mesh_settings(self):
+        return self._parent._edb_sim_setup_info.SimulationSettings.AdvancedMeshSettings
+
+    @property
+    def layer_snap_tol(self):
+        """Layer snap tolerance. Attempt to align independent stackups in the mesher.
+
+        Returns
+        -------
+        str
+
+        """
+        return self._advanced_mesh_settings.LayerSnapTol
+
+    @layer_snap_tol.setter
+    def layer_snap_tol(self, value):
+        self._advanced_mesh_settings.LayerSnapTol = value
+        self._parent._update_setup()
+
+    @property
+    def mesh_display_attributes(self):
+        """Mesh display attributes. Set color for mesh display (i.e. ``"#000000"``).
+
+        Returns
+        -------
+        str
+        """
+        return self._advanced_mesh_settings.MeshDisplayAttributes
+
+    @mesh_display_attributes.setter
+    def mesh_display_attributes(self, value):
+        self._advanced_mesh_settings.MeshDisplayAttributes = value
+        self._parent._update_setup()
+
+    @property
+    def replace_3d_triangles(self):
+        """Whether to turn on replace 3D triangles.
+
+        Returns
+        -------
+        bool
+            ``True`` if replace 3D triangles is used, ``False`` otherwise.
```

Comment:
```suggestion
        """Whether to turn on replace 3D triangles.

        Returns
        -------
        bool
            ``True`` if replace 3D triangles is used, ``False`` otherwise.
```
```suggestion
        """Whether 3D triangles are replaced.

        Returns
        -------
        bool
            ``True`` if 3D triangles are replaced, ``False`` otherwise.
```